tests/gnrc_dhcpv6_client/tests-as-root/01-run.py

```diff
@@ -23,17 +28,33 @@ def testfunc(child):
     child.expect(r"Iface\s+\d+")
     child.expect(r"inet6 addr:\s+fe80:[0-9a-f:]+\s+scope: link")
 
-    global_addr_1, global_addr_2 = extract_global_addresses(child)
+    global_addr_1 = extract_global_address(child)
+    global_addr_2 = extract_global_address(child)
+    assert_unique_elements(global_addr_1, global_addr_2)
+
+    global_pfx_1 = extract_global_prefix(child)
+    global_pfx_2 = extract_global_prefix(child)
+    assert_unique_elements(global_pfx_1, global_pfx_2)
+
+    test_data = [(global_addr_1, global_pfx_1), (global_addr_2, global_pfx_2)]
+
+    perform_test(test_data, check_ia_na_addr)
+    perform_test(test_data, check_ia_pd_addr)
 
-    global_pfx = extract_global_prefix(child)
 
-    test_global_addrs(global_addr_1, global_addr_2, global_pfx)
+def perform_test(test_data, test_function):
+    result = [
+        (addr, prefix) for addr, prefix in test_data if test_function(addr, prefix)
+    ]
+    assert len(result) == 1
```

Comment:
Someone likes to flex there list comprehension game? I find this hard to understand without any comments.

Reply:
I now replaced it with a regular for-loop, I hope that's a bit more readable :)

Reply:
I originally intended to use something like this:

```python
def perform_test(test_data, test_function):
    """Assert that only one address matches the given test function"""
    assertion = lambda address, prefix: test_function(address, prefix)
    result = filter(assertion, test_data)
    assert len(result) == 1
```

But I am not sure if a lambda function is very readable either.

Reply:
I think the function parameter makes it not very readable either TBH... Regardless of list comprehension or spelled out for-loop. IMHO the test should be as flat as possible (unless large parts are repeated multiple times). The current way seems to me to be as general as possible, where generality is not really needed (our set has two entries) and in this case makes it also harder to read. In my opinion

```py
assert check_ia_na_addr(global_addr_1, global_pfx_1) or \
       check_ia_pd_addr(global_addr_1, global_pfx_1)
assert check_ia_na_addr(global_addr_2, global_pfx_2) or \
       check_ia_pd_addr(global_addr_2, global_pfx_2)
```

makes it the clearest what you are actually trying to test. It might not be as spiffy, but at least one does not have to study your code first to understand the test ;-).

Reply:
Alternatively, if you want to keep the dict structure

```assert check_ia_na_addr(**test_data[0]) or \
       check_ia_pd_addr(**test_data[0])
assert check_ia_na_addr(**test_data[1]) or \
       check_ia_pd_addr(**test_data[1])
```

would also work for me.

Reply:
> ```python
> assert check_ia_na_addr(global_addr_1, global_pfx_1) or \
>        check_ia_pd_addr(global_addr_1, global_pfx_1)
> assert check_ia_na_addr(global_addr_2, global_pfx_2) or \
>        check_ia_pd_addr(global_addr_2, global_pfx_2)
> ```

I was also thinking a solution like this in the meantime. I added your code in 9af88ccac28aa5c35a69110c09c5845c595acf68. I agree that it is better to keep it simple for now :)